unlock-app/src/components/interface/connect/Wallet.tsx

```diff
@@ -1,213 +1,44 @@
-          setError(
-            'email',
-            {
-              type: 'value',
-              message: error.message,
-            },
-            {
-              shouldFocus: true,
-            }
-          )
-        }
-      }
-    }
-  }
-
-  return (
-    <div>
-      <form className="grid gap-4" onSubmit={handleSubmit(onSubmit)}>
-        <Input
-          type="email"
-          autoComplete="email"
-          placeholder={email ? email : 'your@email.com'}
-          error={errors.email?.message}
-          {...register('email', {
-            required: {
-              value: true,
-              message: 'Email is required',
-            },
-          })}
-          actions={
-            <Button
-              type="submit"
-              disabled={!watch('email')}
-              variant="borderless"
-              loading={isSubmitting || isLoadingUserExists}
-              className="p-2.5"
-            >
-              Continue
-            </Button>
-          }
-        />
-      </form>
-    </div>
-  )
-}
-
-export const ConnectWallet = ({
-  injectedProvider,
-  shoudOpenConnectModal = false,
-  checkoutService,
-}: ConnectWalletProps) => {
-  const { email } = useAuth()
-  const [userEmail, setUserEmail] = useState<string | undefined>(
-    email || undefined
-  )
+export const ConnectWallet = () => {
+  // https://docs.privy.io/guide/react/wallets/external/#connect-or-create
+  const { login, logout, ready, authenticated, user } = usePrivy()
+  const { signIn } = useSIWE()
+  const { wallets } = useWallets()
 
-  const [isEmailLoading, setIsEmailLoading] = useState<boolean>(false)
-
-  const { data: userType } = useQuery({
-    queryKey: ['userAccountType', userEmail],
-    queryFn: async () => {
-      setIsEmailLoading(true)
-      const result = await locksmith.getUserAccountType(userEmail as string)
-      const userAccountType = result.data.userAccountType as UserAccountType[]
-      setIsEmailLoading(false)
-      return userAccountType
-    },
-    enabled: !!userEmail,
-  })
-
-  const { authenticateWithProvider } = useAuthenticate({ injectedProvider })
-  const [recentlyUsedProvider] = useLocalStorage(RECENTLY_USED_PROVIDER, null)
-  const [isConnecting, setIsConnecting] = useState('')
-
-  const createOnConnectHandler = (provider: any) => {
-    const handler: MouseEventHandler<HTMLButtonElement> = async (event) => {
-      event.preventDefault()
-      setIsConnecting(provider)
-      await authenticateWithProvider(provider)
-      setIsConnecting('')
-    }
-    return handler
-  }
-
-  const verifyAndSetEmail = async (email: string) => {
-    if (email) {
-      setUserEmail(email)
-    }
-  }
+  useEffect(() => {
+    console.log('ready', ready)
+    console.log('render login')
+    console.log('authenticated', authenticated)
+    console.log('user', user)
+    login()
+  }, [])
 
   useEffect(() => {
-    if (!email) {
-      setUserEmail('')
+    if (authenticated) {
```

Comment:
once the user connects and signs via privy, we immediately call our signIn function which triggers `loginWithPrivy`